go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -25,7 +25,7 @@ class GoModUpdater
           /fatal: The remote end hung up unexpectedly/,
           /repository '.+' not found/,
           # (Private) module could not be fetched
-          /go: .*: git fetch .*: exit status 128/m.freeze,
+          /go: .*: git (fetch|ls-remote) .*: exit status 128/m.freeze,
```

Comment:
In go 1.17 two of the test cases from https://github.com/dependabot/dependabot-core/pull/4130 changed:
| error | go 1.16 | go 1.17 |
| ----- | ------- | ------- |
| direct dependency unreachable | go: github.com/dependabot-fixtures/go-modules-private@v0.0.1: reading github.com/dependabot-fixtures/go-modules-private/go.mod at revision v0.0.1: unknown revision v0.0.1 | go: github.com/dependabot-fixtures/go-modules-private@v0.0.1: reading github.com/dependabot-fixtures/go-modules-private/go.mod at revision v0.0.1: git ls-remote -q origin in /home/dependabot/go/pkg/mod/cache/vcs/51c9292b9f45a35312c3863da5428630dd2a800586fd865c7065132199b41694: exit status 128: fatal: could not read Username for 'https://github.com': terminal prompts disabled Confirm the import path was entered correctly. If this is a private repository, see https://golang.org/doc/faq#git_https for additional information. |
| sub dependency unreachable | go: github.com/dependabot-fixtures/go-modules-private-dependency@v0.0.1 requires github.com/dependabot-fixtures/go-modules-private@v0.0.1: reading github.com/dependabot-fixtures/go-modules-private/go.mod at revision v0.0.1: unknown revision v0.0.1 | go: github.com/dependabot-fixtures/go-modules-private-dependency@v0.0.1 requires github.com/dependabot-fixtures/go-modules-private@v0.0.1: reading github.com/dependabot-fixtures/go-modules-private/go.mod at revision v0.0.1: git ls-remote -q origin in /home/dependabot/go/pkg/mod/cache/vcs/51c9292b9f45a35312c3863da5428630dd2a800586fd865c7065132199b41694: exit status 128: fatal: could not read Username for 'https://github.com': terminal prompts disabled Confirm the import path was entered correctly. If this is a private repository, see https://golang.org/doc/faq#git_https for additional information. |